core/src/main/java/org/apache/accumulo/core/security/Authorizations.java

```diff
@@ -43,11 +43,11 @@
 public class Authorizations implements Iterable<byte[]>, Serializable, AuthorizationContainer {
 
   private static final long serialVersionUID = 1L;
-  private static final Set<ByteSequence> EMPTY_AUTH_SET = Collections.emptySet();
-  private static final List<byte[]> EMPTY_AUTH_LIST = Collections.emptyList();
+  private static final HashSet<ByteSequence> EMPTY_AUTH_SET = new HashSet<>();
+  private static final ArrayList<byte[]> EMPTY_AUTH_LIST = new ArrayList<>();
```

Comment:
These changes will make these mutable, and they could leak out, have stuff added to them, and then no longer be empty.

I'm not sure what's motivating this particular change, but it seems like it might have been okay before? I'm not sure why this one would be a problem for JDK 21.

Reply:
It would be good to keep those immutable.

Reply:
The error was `non-transient instance field of a serializable class declared with a non-serializable type`

Reply:
Could we delete those two variable and wherever they are used use Set.of() and List.of() instead?

Reply:
`Set.of()` and `List.of()` have issues with `.contains(null)`, so can be annoying (they throw an exception rather than return false). They might work here if they are serializable, but I'm not sure.

I'm very confused about the error, because these should be static fields, not instance fields, so shouldn't be serialized with the instance anyway. But, if I'm confused about that, it seems the simplest fix is to just mark them transient, so they are excluded from serialization. They are already initialized when the class is loaded, so it shouldn't be an issue.

Reply:
So, it wasn't the static fields that were an issue. It was the instance fields. But then we return the static fields in some cases. I may have over-corrected on this one.

Reply:
> However, I'm very confused about the error, because these should be static fields, not instance fields, so shouldn't be serialized with the instance anyway. 

I was wondering about that, if you find out the reasoning let me know.

> We could certainly get rid of the variables and just use the emptyList or emptySet instead, though.

That sounds good, no surprises with null.